### PR TITLE
Revise "Managing Chronograf security" doc

### PR DESCRIPTION
**Pull request by pierwill**
This doc is very large. I've done a reorganization of the material, and edited through the OAuth introduction. I'll work on editing the specific OAuth provider instructions and the TLS instructions in later PRs.

### CODE REVIEW

File: content/chronograf/v1.7/administration/managing-security.md [hunk 1]
```diff
@@ -9,50 +9,47 @@ menu:
     parent: Administration
 ---
 
-**On this page**
+Chronograf supports authentication and authorization with [OAuth 2.0](https://oauth.net/) and can be configured to use TLS/HTTPS.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Notice we have 2 OAuth 2.0 link going to diff sites. Does it make more sense to keep both links and clarify the distinction, or use one link or the other?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 2]
```diff
@@ -9,50 +9,47 @@ menu:
...
+To enhance security, follow the instructions on this page to configure both features.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: To enhance security, configure Chronograf to authenticate and authorize with OAuth 2.0 and TLS/HTTPS:
- Configure OAuth:
   a. Generate secret token
   b. Set configurations for your OAuth provider
- Configure TLS...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 3]
```diff
@@ -9,50 +9,47 @@ menu:
...
-  * [Configuring Heroku authentication](#configuring-heroku-authentication)
-  * [Configuring Okta authentication](#configuring-okta-authentication)
-  * [Configuring GitLab authentication](#configuring-gitlab-authentication)
-  * [Configuring Generic authentication](#configuring-generic-authentication)
-* [Configuring authentication duration](#configuring-authentication-duration)
-* [Configuring TLS (Transport Layer Security) and HTTPS](#configuring-tls-transport-layer-security-and-https)
-  - [Testing with self-signed certificates](#testing-with-self-signed-certificates)
+* [Configure OAuth 2.0](#configure-oauth-2-0)
+  - [OAuth 2.0 providers](#oauth-2-0-providers)
+* [Configure TLS (Transport Layer Security) and HTTPS](#configuring-tls-transport-layer-security-and-https)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: (#configure-tls

File: content/chronograf/v1.7/administration/managing-security.md [hunk 4]
```diff
@@ -9,50 +9,47 @@ menu:
...
-## Chronograf security
+## Configure OAuth 2.0
 
-To enhance security, configure both of the following options for Chronograf:
+> After configuring OAuth 2.0, the Chronograf Admin tab becomes visible.
+> You can then set up [multiple organizations](https://docs.influxdata.com/chronograf/latest/administration/managing-organizations/)
+> and [users](https://docs.influxdata.com/chronograf/latest/administration/managing-influxdb-users/).
 
-* OAuth 2.0 providers with JWT tokens
+Chronograf can use [OAuth 2.0](https://tools.ietf.org/html/rfc6749) providers and [JWT (JSON Web Token)](https://tools.ietf.org/html/rfc7519) tokens
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Configure Chronograf to use an OAuth 2.0 provider and JWT (JSON Web Token) token to authenticate a Chronograf user and authorize permissions for their assigned roles?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 5]
```diff
@@ -9,50 +9,47 @@ menu:
...
+to require authorization and authentication of Chronograf users and enable role-based access controls.
 
->**Note** After configuring OAuth 2.0, the Chronograf Admin tab becomes visible, and you can set up [multiple organizations] (https://docs.influxdata.com/chronograf/latest/administration/managing-organizations/) and [users](https://docs.influxdata.com/chronograf/latest/administration/managing-influxdb-users/).
+### Generate a Token Secret
 
-* TLS (Transport Layer Security) for HTTPS
+To configure any of the supported OAuth 2.0 providers to work with Chronograf,
+you must also configure the `TOKEN_SECRET` environment variable (or command line option), which is used for generating and validating JWT tokens.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: you must configure......to generate a JWT signature that validates the JWT token?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 6]
```diff
@@ -9,50 +9,47 @@ menu:
...
+Chronograf will use this secret to generate the JWT Signature for all access tokens.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 1. Set the token value...

diff between JWT and JWKS signature?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 7]
```diff
@@ -9,50 +9,47 @@ menu:
...
-
-## OAuth 2.0 providers with JWT tokens
-
-[OAuth 2.0](https://tools.ietf.org/html/rfc6749) providers and [JWT (JSON Web Token)](https://tools.ietf.org/html/rfc7519) tokens are used in Chronograf to provide the authorization and authentication of Chronograf users and role-based access control.
+```
+TOKEN_SECRET=<mysecret>
+```
 
-To configure any of the supported OAuth 2.0 providers to work with Chronograf, you must also configure the `TOKEN_SECRET` environment variable (or command line option), which is used for generating and validating JWT tokens.
+Generate a secret with OpenSSL by running this command:
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 2. Generate...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 8]
```diff
@@ -9,50 +9,47 @@ menu:
...
 
-**To set the TOKEN_SECRET option:**
+```
+openssl rand -base64 32
+```
 
-Set the value of the TOKEN_SECRET environment variable to a secure, arbitrary string. Chronograf will use this secret to generate the JWT Signature for all access tokens.
+> ***InfluxEnterprise clusters:*** If you are running multiple Chronograf servers in a high availability configuration,
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 3. If you're running multiple Chronograf servers...,

File: content/chronograf/v1.7/administration/managing-security.md [hunk 9]
```diff
@@ -9,50 +9,47 @@ menu:
...
+> set the `TOKEN_SECRET` environment variable on each server to ensure that users can stay logged in.
 
-**Example:** `TOKEN_SECRET=Super5uperUdn3verGu355!`
+### JWKS Signature Verification
 
-**JWKS Signature Verification**
-If the provider implements OpenID Connect with RS256 signatures (as Microsoft AD FS does), you need to enable `id_token` support and provide a JWKS document (holding the certificate chain), to validate the RSA signatures against. This certificate chain is regularly rolled over (when the certificates expire), so it is fetched from the JWKS_URL on demand.
+If the OAuth provider implements OpenID Connect with RS256 signatures (as Microsoft AD FS does),
+you need to enable `id_token` support and provide a JWKS document (holding the certificate chain), to validate the RSA signatures against.
+This certificate chain is regularly rolled over (when the certificates expire), so it is fetched from the JWKS_URL on demand.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: When a certificate (in the certificate chain expires), the certificate is automatically fetched from the JWKS_URL on demand.
or
When a certificate expires, OAuth automatically fetches the certificate from the JWKS_URL on demand.

File: content/chronograf/v1.7/administration/managing-security.md [hunk 10]
```diff
@@ -61,13 +58,12 @@ export USE_ID_TOKEN=true
...
 ```
 
-> ***InfluxEnterprise clusters:*** If you are running multiple Chronograf servers in a high availability configuration, set the `TOKEN_SECRET` environment variable on each server to ensure that users can stay logged in.
-
-## OAuth 2.0 providers
+### OAuth 2.0 providers
 
-OAuth 2.0 authorization and authentication in Chronograf require you to specify configuration options for OAuth 2.0 authentication providers you want to use.
+To enable OAuth 2.0 authorization and authentication in Chronograf,
+you must set configuration options that are specific for the OAuth 2.0 authentication provider you want to use.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Set configuration options for the provider you want to use:
- GitHub...

If you haven't already, you must first [generate a secret token]...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 11]
```diff
@@ -76,16 +72,18 @@ Configuration steps, including required and optional configuration options, for
 * [Okta](#configuring-okta-authentication)
 * [Generic](#configuring-generic-authentication)
 
-> ***Note:*** Each of the following OAuth 2.0 provider configurations require the `TOKEN_SECRET` you created in the previous section.
+> These configurations require a `TOKEN_SECRET`, described [above](#configure-oauth-2).
 
 ### Configuring GitHub authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ### Configure GitHub...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 12]
```diff
@@ -76,16 +72,18 @@ Configuration steps, including required and optional configuration options, for
...
 
-Chronograf supports using the [GitHub OAuth 2.0 authentication](https://developer.github.com/apps/building-oauth-apps/) to request authorization and provide authentication. To use GitHub authentication, you need to register a GitHub application and use the assigned Client ID and Client Secret.
-
+Chronograf supports using the [GitHub OAuth 2.0 authentication](https://developer.github.com/apps/building-oauth-apps/)
+to request authorization and provide authentication.
+To use GitHub authentication, you need to register a GitHub application and use the assigned Client ID and Client Secret.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 1. Register a GitHub application and use the assigned Client ID and Client Secret.
2. Set the following required values:

File: content/chronograf/v1.7/administration/managing-security.md [hunk 13]
```diff
@@ -159,28 +159,30 @@ The following steps will guide you in configuring Chronograf to support GitHub O
...
-> When logging in for the first time, make sure to grant access to the organization you configured. The OAuth application can only see membership in organizations it has been granted access to.
-
+> When logging in for the first time, make sure to grant access to the organization you configured.
+> The OAuth application can only see membership in organizations it has been granted access to.
 
 ### Configuring Google authentication
 
-Chronograf supports using the [Google OAuth 2.0 authentication provider](https://developers.google.com/identity/protocols/OAuth2) to request authorization and provide authentication. To use Google authentication, you need to register a Google application and use the assigned Client ID and Client Secret, as well as specify a Public URL.
+Chronograf supports using the [Google OAuth 2.0 authentication provider](https://developers.google.com/identity/protocols/OAuth2) to request authorization and provide authentication.
+To use Google authentication, you need to register a Google application and use the assigned Client ID and Client Secret, as well as specify a Public URL.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 1. [Register your GitHub application](<link>) and use the assigned Client ID and Client Secret and specify a public URL.
2. Create the following required Chronograf env....:
    -GOOG...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 14]
```diff
@@ -159,28 +159,30 @@ The following steps will guide you in configuring Chronograf to support GitHub O
...
 
 #### Overview
 
-Chronograf supports the use of the [Google OAuth 2.0 protocol](https://developers.google.com/identity/protocols/OAuth2) for authentication and authorization. The steps below guide you in creating the following required Chronograf environment variables (or corresponding command line options):
+Chronograf supports the use of the [Google OAuth 2.0 protocol](https://developers.google.com/identity/protocols/OAuth2) for authentication and authorization.
+The steps below guide you in creating the following required Chronograf environment variables (or corresponding command line options):
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: I'd prob drop overview section/line 184-185 - and drop this same (related) content in all provider sections...content varies slightly in some sections and not included in others

File: content/chronograf/v1.7/administration/managing-security.md [hunk 15]
```diff
@@ -242,15 +243,17 @@ export GOOGLE_DOMAINS=biffspleasurepalance.com,savetheclocktower.com
 
 ### Configuring Auth0 authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ### Configure AuthO...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 16]
```diff
@@ -242,15 +243,17 @@ export GOOGLE_DOMAINS=biffspleasurepalance.com,savetheclocktower.com
...
 
-[Auth0](https://auth0.com/) implements identity protocol support for OAuth 2.0. See [OAuth 2.0](https://auth0.com/docs/protocols/oauth2) for details about the Auth0 implementation.
+[Auth0](https://auth0.com/) implements identity protocol support for OAuth 2.0.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 1. Create an AuthO account and register...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 17]
```diff
@@ -309,7 +312,8 @@ Once your application has been created, those two values should be inserted into
 #### Optional Heroku organizations
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Is Heroku an optional subset of AuthO? In OAuth 2.0 providers sections, it reads like it's on same level.

File: content/chronograf/v1.7/administration/managing-security.md [hunk 18]
```diff
@@ -320,7 +324,7 @@ export HEROKU_ORGS=hill-valley-preservation-sociey,the-pinheads
 ```
 ### Configuring Okta authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ### Configure Okta...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 19]
```diff
@@ -320,7 +324,7 @@ export HEROKU_ORGS=hill-valley-preservation-sociey,the-pinheads
...
 
-[Okta](https://developer.okta.com/) is a popular, OAuth 2.0 compliant authorization and authentication provider that can be used with Chronograf to allow access based on granted scopes and permissions.
+[Okta](https://developer.okta.com/) is a popular OAuth 2.0 compliant authorization and authentication provider that can be used with Chronograf to allow access based on granted scopes and permissions.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: 1. Create an Okta account and register...
2. Set up env variables for Okta?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 20]
```diff
@@ -355,7 +359,8 @@ export HEROKU_ORGS=hill-valley-preservation-sociey,the-pinheads
 Your users should now be able to sign into Chronograf using the new Okta provider.
 
 #### Configuring GitLab authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ### Configure GitLab...
3rd level header?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 21]
```diff
@@ -355,7 +359,8 @@ export HEROKU_ORGS=hill-valley-preservation-sociey,the-pinheads
...
-[GitLab](https://gitlab.com/) is a popular, OAuth 2.0 compliant authorization and authentication provider that can be used with Chronograf to allow access based on granted scopes and permissions.
+
+[GitLab](https://gitlab.com/) is a popular OAuth 2.0 compliant authorization and authentication provider that can be used with Chronograf to allow access based on granted scopes and permissions.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: are other providers less popular? maybe:
1. Set the env variables:

File: content/chronograf/v1.7/administration/managing-security.md [hunk 22]
```diff
@@ -471,47 +475,54 @@ TOKEN_SECRET="ZNh2N9toMwUVQxTVEe2ZnnMtgkh3xqKZ"
 
 > _**Note:**_ Do not use special characters for the GENERIC_CLIENT_ID as AD FS will split strings here, finally resulting in an identifier mismatch.
 
-
 ### Configuring authentication duration
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ### Configure

File: content/chronograf/v1.7/administration/managing-security.md [hunk 23]
```diff
@@ -471,47 +475,54 @@ TOKEN_SECRET="ZNh2N9toMwUVQxTVEe2ZnnMtgkh3xqKZ"
...
 
-Chronograf includes command line and environment variable options for configuring TLS (Transport Layer Security) certificates and key files. Use of the TLS cryptographic protocol provides server authentication, data confidentiality, and data integrity. When configured, users can use HTTPS to securely communicate with your Chronograf applications.
+Chronograf includes command line and environment variable options for configuring TLS (Transport Layer Security) certificates and key files.
+Use of the TLS cryptographic protocol provides server authentication, data confidentiality, and data integrity.
+When configured, users can use HTTPS to securely communicate with your Chronograf applications.
 
-> ***Note:*** Using HTTPS helps guard against nefarious agents sniffing the JWTand using it to spoof a valid user against the Chronograf server.
+> ***Note:*** Using HTTPS helps guard against nefarious agents sniffing the JWT and using it to spoof a valid user against the Chronograf server.
 
 ### Configuring TLS for Chronograf
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ### Configure

File: content/chronograf/v1.7/administration/managing-security.md [hunk 24]
```diff
@@ -9,12 +9,12 @@ menu:
...
 ---
 
-Chronograf supports authentication and authorization with [OAuth 2.0](https://oauth.net/) and can be configured to use TLS/HTTPS.
-To enhance security, follow the instructions on this page to configure both features.
+To enhance security, configure Chronograf to authenticate and authorize with [OAuth 2.0](https://oauth.net/) and TLS/HTTPS.
 
 * [Configure OAuth 2.0](#configure-oauth-2-0)
-  - [OAuth 2.0 providers](#oauth-2-0-providers)
-* [Configure TLS (Transport Layer Security) and HTTPS](#configuring-tls-transport-layer-security-and-https)
+  - [Generate a Token Secret](#generate-a-token-secret)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: if folks use the generated token when they configure their OAuth provider, I'd prob indicate the order w an ordered list

File: content/chronograf/v1.7/administration/managing-security.md [hunk 25]
```diff
@@ -9,12 +9,12 @@ menu:
...
+  - [Set configurations for your OAuth provider](#oauth-2-0-providers)
+* [Configure TLS (Transport Layer Security) and HTTPS](#confige-tls-transport-layer-security-and-https)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: (#configure

File: content/chronograf/v1.7/administration/managing-security.md [hunk 26]
```diff
@@ -358,7 +358,7 @@ export HEROKU_ORGS=hill-valley-preservation-sociey,the-pinheads
 
 Your users should now be able to sign into Chronograf using the new Okta provider.
 
-#### Configuring GitLab authentication
+### Configuring GitLab authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Will, maybe start header w "Configure" to keep parallel w other headers in this topic

File: content/chronograf/v1.7/administration/managing-security.md (hunk 1 above)
Comment (pierwill) [reply content/chronograf/v1.7/administration/managing-security.md]: I think I'll link to the oauth.net site first, then add a note about "for details on oauth and jwt, see rfc x and y."

**Comment by pierwill** [comment]
Will address remaining feedback in future PRs.

**Pull request merged**